contracts/src/bin/deploy.rs

```diff
@@ -32,6 +32,10 @@ async fn run() -> Result<()> {
     log::info!("checking connection to local test node {}", NODE_URL);
     wait_for_node(&web3).await?;
 
+    let network_id = web3.net().version().await.expect("get network ID failed");
+    write_network_id(&network_id)?;
+    log::info!("connected to test network {}", network_id);
```

Comment:
Do we really want to explicitly state "test" network here? Is this a test file? Maybe we should call this file `deploy_test.rs` if this is the case.

Reply:
This is the deploy script for deploying contracts for E2E testing. I'm not a fan of adding `_test` postscript to this file.